core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.spec.ts

```diff
@@ -0,0 +1,920 @@
+      routerService = $injector.get('RouterService');
+      translationLanguageService = $injector.get('TranslationLanguageService');
+      translationTabActiveContentIdService = $injector.get(
+        'TranslationTabActiveContentIdService');
+      translationTabActiveModeService = $injector.get(
+        'TranslationTabActiveModeService');
+
+      spyOn(stateEditorService, 'getActiveStateName').and.returnValue(
+        'Introduction');
+      explorationStatesService.init(explorationState2);
+      stateRecordedVoiceoversService.init('Introduction',
+        recordedVoiceoversObjectFactory.createFromBackendDict(
+          recordedVoiceovers));
+
+      $scope = $rootScope.$new();
+      ctrl = $componentController('stateTranslation', {
+        $filter: $filter,
+        $rootScope: $rootScope,
+        $scope: $scope,
+        CkEditorCopyContentService: ckEditorCopyContentService,
+        StateEditorService: stateEditorService
+      }, {
+        isTranslationTabBusy: false
+      });
+      ctrl.$onInit();
+    }));
+
+    it('should activate feedback tab with default outcome when' +
+      ' clicking on tab', function() {
+      spyOn(translationTabActiveContentIdService, 'setActiveContentId');
+      $scope.onTabClick('feedback');
+
+      expect($scope.isActive('feedback')).toBe(true);
+      expect($scope.isDisabled('feedback')).toBe(false);
+      expect(translationTabActiveContentIdService.setActiveContentId)
+        .toHaveBeenCalledWith('default_outcome');
+    });
+  });
+
+  describe('when state is linear', function() {
+    beforeEach(angular.mock.inject(function($injector, $componentController) {
+      $rootScope = $injector.get('$rootScope');
+      explorationStatesService = $injector.get('ExplorationStatesService');
+      routerService = $injector.get('RouterService');
+      translationLanguageService = $injector.get('TranslationLanguageService');
+      translationTabActiveContentIdService = $injector.get(
+        'TranslationTabActiveContentIdService');
+      translationTabActiveModeService = $injector.get(
+        'TranslationTabActiveModeService');
+
+      spyOn(stateEditorService, 'getActiveStateName').and.returnValue(
+        'Introduction');
+      explorationStatesService.init(explorationState3);
+      stateRecordedVoiceoversService.init('Introduction',
+        recordedVoiceoversObjectFactory.createFromBackendDict(
+          recordedVoiceovers));
+
+      $scope = $rootScope.$new();
+      ctrl = $componentController('stateTranslation', {
+        $rootScope: $rootScope,
+        $scope: $scope,
+        CkEditorCopyContentService: ckEditorCopyContentService,
+        StateEditorService: stateEditorService
+      }, {
+        isTranslationTabBusy: false
+      });
+      ctrl.$onInit();
+    }));
+
+    it('should evaluate hint tab as enabled', function() {
+      expect($scope.isDisabled('content')).toBe(false);
+    });
+
+
+    it('should evaluate feedback tab as disable', function() {
```

Comment:
It's not possible to toggle these values because they depend on the state's property `is_linear` that is not variable after the component is initialized.

Reply:
Ah ok.. Thanks for the explanation